vault/token_store.go

```diff
@@ -530,6 +601,83 @@ func (ts *TokenStore) revokeTreeSalted(saltedId string) error {
 	return nil
 }
 
+func (ts *TokenStore) lookupByAccessorID(accessorID string) (string, error) {
+	entry, err := ts.view.Get(lookupPrefix + ts.SaltID(accessorID))
+	if err != nil {
+		return "", fmt.Errorf("failed to read index using accessor ID: %s", err)
+	}
+	if entry == nil {
+		return "", &StatusBadRequest{Err: "invalid accessor ID"}
+	}
+
+	return string(entry.Value), nil
+}
+
+// handleLookupAccessor handles the auth/token/lookup-accessor path for returning
+// the properties of the token associated with the accessor ID
+func (ts *TokenStore) handleLookupAccessor(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	accessorID := data.Get("accessor_id").(string)
+	if accessorID == "" {
+		return nil, &StatusBadRequest{Err: "missing accessor_id"}
+	}
+
+	tokenID, err := ts.lookupByAccessorID(accessorID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Prepare the field data required for a lookup call
+	d := &framework.FieldData{
+		Raw: map[string]interface{}{
+			"token": tokenID,
+		},
+		Schema: map[string]*framework.FieldSchema{
+			"token": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Token to lookup",
+			},
+		},
+	}
+	resp, err := ts.handleLookup(req, d)
+	if err != nil {
+		return nil, err
+	}
+	if resp == nil {
+		return nil, fmt.Errorf("failed to lookup the token")
+	}
+	if resp.IsError() {
+		return resp, nil
+
+	}
+
+	// Remove the token ID from the response
+	if resp.Data != nil && resp.Data["id"] != "" {
```

Comment:
`resp.Data` is a `map[string]interface{}` so checking "id" should be for `nil`, I think...I'd not even check and just unilaterally set it to the empty string. If there's a data field we can assume it has the token properties.